go/arrow/flight/flightsql/driver/utils.go

```diff
@@ -60,44 +60,135 @@ func (g grpcCredentials) RequireTransportSecurity() bool {
 func fromArrowType(arr arrow.Array, idx int) (interface{}, error) {
 	switch c := arr.(type) {
 	case *array.Boolean:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return c.Value(idx), nil
 	case *array.Float16:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return float64(c.Value(idx).Float32()), nil
```

Comment:
should a float16 return a `float64`? or should it return `float16.Num`? I would lean towards the latter